#
# base patch: meta/recipes-devtools/perl/perl/MM_Unix.pm.patch
# base branch: master
# base commit: dbda297fd91aab2727f7a69d3b7d3a32ad4261d2
#
# Patch has been updated to work with Debian source code
#
# Upstream-Status:Pending
#
# The perl-native should be relocatable while re-use sstate-cache from
# another build, especially the original build was renamed/removed.
#
# To get the MakeMaker.pm works in this case, we need perl wrapper here
# instead of real perl binary.
#
# Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
===================================================================
--- perl-5.12.3/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm.orig	2011-08-12 16:07:30.000000000 +0800
+++ perl-5.12.3/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm	2011-08-12 16:08:56.000000000 +0800
@@ -1019,6 +1019,9 @@
     }
 
     foreach my $name (@$names){
+        # Getting MakeMaker.pm use perl wrapper instead of 'perl.real' directly
+        $name =~ s/perl\.real/perl/ if ($name =~ /perl\.real/);
+
         my ($abs, $use_dir);
         if ($self->file_name_is_absolute($name)) {     # /foo/bar
             $abs = $name;
